libtransact/src/execution/executer_internal.rs

```diff
@@ -575,23 +576,27 @@ mod tests {
             }
         }
 
-        // Process the ExecutionTask and return an ExecutionResult.
+        // Process the ExecutionTask and return an TransactionExecutionResult.
 
         while let Ok(event) = receiver.try_recv() {
             if let ExecutionCommand::Event(execution_event) = event {
                 let (result_sender, task) = *execution_event;
-                let transaction_status = TransactionStatus::Valid;
-                let execution_result = ExecutionResult {
-                    transaction_id: task.pair().transaction().header_signature().to_string(),
-                    status: transaction_status,
-                };
+
+                let execution_result = TransactionExecutionResult::Valid(
+                    TransactionReceiptBuilder::new()
```

Comment:
I think the scheduler should be putting together this receipt, since it is the originator and destroyer of a Context.  The Execution Adapters can't do this without having the context manager lifecycle methods.

Reply:
Fixed